designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/ListCard.kt

```diff
@@ -0,0 +1,320 @@
+ * @param textAvatar style must be TEXT, will show a single character as avatar
+ * @param imagePainter style must be IMAGE, will display an image as avatar
+ * @param metadataAvatar style must be METADATA, composable should be DHIS2 [MetadataAvatar]
+ * @param modifier allows a modifier to be passed externally
+ */
+@Composable
+fun ListAvatar(
+    textAvatar: String? = null,
+    imagePainter: Painter = provideDHIS2Icon("dhis2_microscope_outline"),
+    metadataAvatar: (@Composable () -> Unit)? = null,
+    style: ListAvatarStyle = ListAvatarStyle.TEXT,
+    onImageClick: (() -> Unit)? = null,
+    modifier: Modifier = Modifier,
+) {
+    when (style) {
+        ListAvatarStyle.TEXT -> {
+            textAvatar?.let {
+                Box(
+                    modifier = modifier
+                        .size(Spacing.Spacing40)
+                        .background(color = SurfaceColor.PrimaryContainer, shape = RoundedCornerShape(Radius.Full)),
+                    contentAlignment = Alignment.Center,
+                ) {
+                    Text(text = textAvatar, color = SurfaceColor.Primary, style = MaterialTheme.typography.titleSmall)
+                }
+            }
+        }
+        ListAvatarStyle.METADATA -> {
+            metadataAvatar?.let {
+                metadataAvatar.invoke()
+            }
+        }
+        ListAvatarStyle.IMAGE -> {
+            Image(
+                painter = imagePainter,
+                contentDescription = "avatarImage",
+                contentScale = ContentScale.Crop,
+                modifier = modifier
+                    .size(Spacing.Spacing40)
+                    .clip(CircleShape)
+                    .clickable(onClick = { onImageClick?.invoke() }),
+            )
+        }
+    }
+}
+
+/**
+ * DHIS2 AdditionalInfoColumn,
+ *  used to display both key value lists, the constant one and the expandable one
+ */
+@Composable
+private fun AdditionalInfoColumn(
+    modifier: Modifier = Modifier,
+    expandableItems: List<AdditionalInfoItem>? = null,
+    constantItems: List<AdditionalInfoItem>,
+    syncProgressItem: AdditionalInfoItem,
+    showLoading: Boolean,
+    expandLabelText: String,
+    shrinkLabelText: String,
+) {
+    val loadingSectionState by remember(showLoading) { mutableStateOf(showLoading) }
+    var sectionState by remember(SectionState.CLOSE) { mutableStateOf(SectionState.CLOSE) }
+
+    var hideableItemList: MutableList<AdditionalInfoItem>
+    val hiddenItemList = mutableListOf<AdditionalInfoItem>()
+    Column(
+        modifier = modifier,
+    ) {
+        if (expandableItems != null && expandableItems.size > 3) {
+            hideableItemList = mutableListOf(expandableItems[0], expandableItems[1], expandableItems[2])
+            KeyValueList(hideableItemList)
+            expandableItems.forEach { item ->
+                if (expandableItems.indexOf(item) > 3) {
+                    hiddenItemList.add(item)
```

Comment:
to create your `hideableItemList` and `hiddenItemList` I think that you can do:
`val hideableItemList = expandableItems.take(3)`
`val hiddenItemList = expandableItems.drop(3)`

Reply:
much cleaner thank you I'll modify it